src/world/scheduler/builder.rs

```diff
@@ -262,6 +262,35 @@ impl WorkloadBuilder {
             Err(err) => panic!("{:?}", err),
         }
     }
+    /// Extends the builder with the following [`WorkloadBuilder`].
+    /// This allows us to collect systems in different builders before joining them together.
+    ///
+    /// [`WorldloadBuilder`]: struct.WorldBuilder.html
+    pub fn extend(&mut self, other: &mut Self) -> &mut Self {
+        let offset_ranges_by = self.borrow_info.len();
+        self.borrow_info.extend(other.borrow_info.drain(..));
+        self.systems.extend(
+            other
+                .systems
+                .drain(..)
+                .map(
+                    |(type_id, type_name, mut borrow_info_range, is_send_sync, system_fn)| {
+                        borrow_info_range.start += offset_ranges_by;
+                        borrow_info_range.end += offset_ranges_by;
+                        (
+                            type_id,
+                            type_name,
+                            borrow_info_range,
+                            is_send_sync,
+                            system_fn,
+                        )
+                    },
+                )
+                .collect::<Vec<_>>(),
```

Comment:
```suggestion
```

This collect might not be necessary. I think I put it here because I was struggling with some other mut issues.